python/lsst/ts/dsm/dsm_csc.py

```diff
@@ -70,37 +71,32 @@ async def close_tasks(self):
 
         This method will remove the telemetry directory if in simulation mode.
         """
-        if self.simulation_mode:
+        if self._requested_simulation_mode:
             if os.path.exists(self.telemetry_directory):
                 shutil.rmtree(self.telemetry_directory)
 
         await super().close_tasks()
 
-    async def configure(self, config):
-        """Send settingsApplied messages.
-
-        Parameters
-        ----------
-        config : `types.SimpleNamespace`
-            The current configuration.
+    def finish_csc_setup(self):
+        """Perform final setup steps for CSC.
         """
-        self.config = config
-        if self.telemetry_directory is None and not self.simulation_mode:
-            self.telemetry_directory = self.config.telemetry_directory
-
-        self.evt_settingsAppliedSetup.set_put(telemetryDirectory=self.telemetry_directory,
-                                              simulationLoopTime=self.simulation_loop_time)
-
-    @staticmethod
-    def get_config_pkg():
-        """Provide the configuration repository / directory.
+        if self._requested_simulation_mode:
+            if (
+                self.telemetry_directory is None
+                or not self.telemetry_directory.startswith("/tmp")
+            ):
+                self.telemetry_directory = tempfile.mkdtemp()
+                self.log.debug(
+                    f"Creating temporary directory: {self.telemetry_directory}"
+                )
+        else:
+            self.telemetry_directory = os.environ.get("DSM_TELEMETRY_DIR")
+            if self.telemetry_directory is None:
+                self.telemetry_directory = "/home/saluser/telemetry"
```

Comment:
Please consider:
```
self.telemetry_directory = os.environ.get("DSM_TELEMETRY_DIR", "/home/saluser/telemetry")
```
Or, better yet, make the default value a module constant defined near the top of the file,
where you can easily see it and document it:
```
# Default telemetry directory if running in real mode and $DSM_TELEMETRY_DIR is not defined
DEFAULT_TELEMETRY_DIR = "/home/saluser/telemetry"
```

Reply:
Done.